python/ray/tune/experiment.py

```diff
@@ -160,16 +160,11 @@ def _register_if_needed(cls, run_object):
 
         if isinstance(run_object, six.string_types):
             return run_object
-        elif isinstance(run_object, types.FunctionType):
-            if run_object.__name__ == "<lambda>":
-                logger.warning(
-                    "Not auto-registering lambdas - resolving as variant.")
-                return run_object
-            else:
-                name = run_object.__name__
-                register_trainable(name, run_object)
-                return name
-        elif isinstance(run_object, type):
+        elif run_object.__name__ == "<lambda>":
+            logger.warning(
+                "Not auto-registering lambdas - resolving as variant.")
```

Comment:
Do we still allow this? I think you need an explicit tune.sample_from() right?

Reply:
oh wait; this is for the `tune.run(lambda x, reporter: reporter(4))` and not for the space.

Reply:
OK actually I get what you mean now; fixed.